sensor/kubernetes/eventpipeline/pipeline.go

```diff
@@ -17,7 +17,20 @@ import (
 	"github.com/stackrox/rox/sensor/kubernetes/listener/resources"
 )
 
-// New instantiates the eventPipeline component
+// New instantiates the eventPipeline component. Sensors pipeline is responsible for the entire lifecycle of a Kubernetes
+// event. From receiving it from the listeners, converting it to storage.Deployment, finding related resources to be
+// reprocessed, and sending events directly to Central or to the detector for processing.
+//
+// Components will communicate with each other using component.ResourceEvent data structure. The pipeline is organized as:
+//
+//	Listener -> Resolver -> Output
+//
+// Each component will write and consume different properties from ResourceEvent, and send the event to the next component in the chain.
+// For an explanation what each property means, check the documentation for component.ResourceEvent.
+//
+// This component introduces a new type of component to sensor:
+// - The event pipeline is a sensor component. I.e. it can send messages to the gRPC stream via the .ResponseC function.
```

Comment:
nit: It may be time to drop the Latin abbreviations, and use more direct English words.

```suggestion
// - The event pipeline is a sensor component. That means, it can send messages to the gRPC stream via the .ResponseC function.
```